spirv_glsl.cpp

```diff
@@ -3542,25 +3542,43 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 			require_extension_internal("GL_ARB_shader_texture_lod");
 	}
 
+	std::string type_prefix = imgtype.image.depth ? "shadow" : "texture";
+
 	if (op == "texture")
-		return join("texture", type);
+		return join(type_prefix, type);
 	else if (op == "textureLod")
 	{
 		if (use_explicit_lod)
-			return join("texture", type, is_legacy_es() ? "LodEXT" : "Lod");
+			return join(type_prefix, type, is_legacy_es() ? "LodEXT" : "Lod");
 		else
-			return join("texture", type);
+			return join(type_prefix, type);
 	}
 	else if (op == "textureProj")
-		return join("texture", type, "Proj");
+		return join(type_prefix, type, "Proj");
 	else if (op == "textureGrad")
-		return join("texture", type, is_legacy_es() ? "GradEXT" : is_legacy_desktop() ? "GradARB" : "Grad");
+		return join(type_prefix, type, is_legacy_es() ? "GradEXT" : is_legacy_desktop() ? "GradARB" : "Grad");
 	else if (op == "textureProjLod")
 	{
 		if (use_explicit_lod)
-			return join("texture", type, is_legacy_es() ? "ProjLodEXT" : "ProjLod");
+			return join(type_prefix, type, is_legacy_es() ? "ProjLodEXT" : "ProjLod");
+		else
+			return join(type_prefix, type);
+	}
+	else if (op == "textureLodOffset")
```

Comment:
Are you sure this exists in legacy GLSL?

Reply:
This one and the other two exist through the `EXT_gpu_shader4` extension. I'll add requiring of that extension when encountering these commands.

Reply:
EXT_gpu_shader4 for these commands is fine.
Do you know anything about GLES though? If it doesn't exist there as an extension to GLES2, please just add a SPIRV_CROSS_THROW for that case.

Reply:
GLES has `EXT_shadow_samplers` which supports at least `shadow2DEXT` and `shadow2DProjEXT`, so these two can be supported. I'll add the throw case for everything else. I'm not aware of any other extension that adds further support here and doing a quick glance through the registered extensions didn't reveal anything either.

Reply:
Sounds good.